overlay/ports/ffmpeg/portfile.cmake

```diff
@@ -40,12 +105,21 @@ if(VCPKG_TARGET_IS_MINGW)
 elseif(VCPKG_TARGET_IS_LINUX)
     string(APPEND OPTIONS " --target-os=linux --enable-pthreads")
 elseif(VCPKG_TARGET_IS_UWP)
-    string(APPEND OPTIONS " --target-os=win32 --enable-w32threads --enable-d3d11va --enable-d3d12va --enable-mediafoundation")
+    string(APPEND OPTIONS " --target-os=win32 --enable-w32threads --enable-mediafoundation")
+    if(NOT ("fedora-ffmpeg-free-safe" IN_LIST FEATURES))
+        string(APPEND OPTIONS " --enable-d3d11va --enable-d3d12va")
+    endif()
 elseif(VCPKG_TARGET_IS_WINDOWS)
-    string(APPEND OPTIONS " --target-os=win32 --enable-w32threads --enable-d3d11va --enable-d3d12va --enable-dxva2 --enable-mediafoundation")
+    string(APPEND OPTIONS " --target-os=win32 --enable-w32threads --enable-mediafoundation")
+    if(NOT ("fedora-ffmpeg-free-safe" IN_LIST FEATURES))
+        string(APPEND OPTIONS " --enable-d3d11va --enable-d3d12va --enable-dxva2")
+    endif()
 elseif(VCPKG_TARGET_IS_OSX)
-    string(APPEND OPTIONS " --target-os=darwin --enable-appkit --enable-avfoundation --enable-coreimage --enable-audiotoolbox --enable-videotoolbox")
-elseif(VCPKG_TARGET_IS_IOS)
+    string(APPEND OPTIONS " --target-os=darwin --enable-appkit")
+    if(NOT ("fedora-ffmpeg-free-safe" IN_LIST FEATURES))
+        string(APPEND OPTIONS " --enable-avfoundation  --enable-coreimage --enable-videotoolbox --enable-audiotoolbox")
+    endif()
+elseif(VCPKG_TARGET_IS_IOS AND NOT ("fedora-ffmpeg-free-safe" IN_LIST FEATURES))
     string(APPEND OPTIONS " --enable-avfoundation --enable-coreimage --enable-videotoolbox")
```

Comment:
Let's add the "fedora-ffmpeg-free-safe" check also to IOS and Android, just in case.

Reply:
I'm only disabling these features because their source is denied. I'd like to blindly disable features here. `jni` and `mediacodec` are allowed by Fedora, so there is no reason to alter it.

Reply:
It looks like Fedora is not consistent for OSs they don't care about. For example there is no need to remove the mediafoundation code, because it is anyway not used in Fedora.

Reply:
Either that, or Apple has a reputation to be quite a pain when it comes to legal action, and the source audit panel decided to play it safe and not include these here because they couldn't ensure they would not be patent ms issues.